lib/customer.rb

```diff
@@ -0,0 +1,48 @@
+require "csv"
+# require_relative 'customers'
+
+
+class Customer
+
+
+  def self.all
+    all_customers = []
+    CSV.read("data/customers.csv").map do |row|
+      address = {}
+      id = row[0].to_i
+      email = row[1]
+      address[:street] = row[2]
+      address[:city] = row[3]
+      address[:state] = row[4]
+      address[:zip] = row[5]
+      new_customer = Customer.new(id, email, address)
+      all_customers << new_customer
+    end
+    return all_customers
+
+  end
+
+  def self.find(id)
+    customers = Customer.all
+
+    customers.each do |c|
+      if c.id == id
+        return c
+      elsif c.id != id
+        return nil
+      end
+    end
+
+
+  end
```

Comment:
This was a nasty and subtle bug (and was the cause of many of your test failures.)

You shouldn't `return nil` on the else branch in your loop.  Doing this means you only ever check the first customer.

You can fix this by moving the `return` out of the loop:

```suggestion
  def self.find(id)
    customers = Customer.all

    customers.each do |c|
      if c.id == id
        return c
    end
    
    return nil
  end
```

Or you can simply use `find` here:

```suggestion
  def self.find(id)
    return Customer.all.find do |c|
      c.id == id
    end
  end
```